src/System.Private.CoreLib/shared/System/ArraySegment.cs

```diff
@@ -172,14 +172,7 @@ public T[] ToArray()
         {
             ThrowInvalidOperationIfDefault();
 
-            if (_count == 0)
```

Comment:
I am not sure whether these generic ones are an improvement. It will increase the max footprint of each `List<T>` or `ArrayList<T>` instantiations.

Reply:
O_o doesn't look good